### PR TITLE
[tune] update params for optimizer in reset_config

### PR DESCRIPTION
**Pull request by hhbyyh**
We should avoid replacing optimizer in `reset_config` since all the states info are discarded. 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: python/ray/tune/examples/pbt_convnet_example.py [hunk 1]
```diff
@@ -52,11 +52,11 @@ def _restore(self, checkpoint_path):
         self.model.load_state_dict(torch.load(checkpoint_path))
 
     def reset_config(self, new_config):
-        del self.optimizer
-        self.optimizer = optim.SGD(
-            self.model.parameters(),
-            lr=new_config.get("lr", 0.01),
-            momentum=new_config.get("momentum", 0.9))
+        for param_group in self.optimizer.param_groups:
+            param_group["lr"] = new_config.get("lr")
```
Comment (richardliaw) [inline python/ray/tune/examples/pbt_convnet_example.py]: what if the new_config does not have `lr` or `momentum`?

Comment (hhbyyh) [reply python/ray/tune/examples/pbt_convnet_example.py]: Do you mean we should use new_config.get("lr", 0.01)? Sure we can always use that. Just want to learn when will new_config does not have lr or momentum. Thanks.

Comment (richardliaw) [reply python/ray/tune/examples/pbt_convnet_example.py]: One situation is if the user does not plan to tune momentum, for example

Comment (hhbyyh) [reply python/ray/tune/examples/pbt_convnet_example.py]: Oh, Do you mean users might want to change the code of this example and try to tune by themselves? Sure. I'll update it with a default value.

File: python/ray/tune/examples/pbt_dcgan_mnist/pbt_dcgan_mnist.py [hunk 2]
```diff
@@ -275,16 +275,11 @@ def _restore(self, checkpoint_dir):
...
-        self.optimizerD = optim.Adam(
-            self.netD.parameters(),
-            lr=new_config.get("netD_lr"),
-            betas=(beta1, 0.999))
-        self.optimizerG = optim.Adam(
-            self.netG.parameters(),
-            lr=new_config.get("netG_lr"),
-            betas=(beta1, 0.999))
+        for param_group in self.optimizerD.param_groups:
+            param_group["lr"] = new_config.get("netD_lr")
```
Comment (richardliaw) [inline python/ray/tune/examples/pbt_dcgan_mnist/pbt_dcgan_mnist.py]:
```suggestion
            param_group["lr"] = new_config.get("netD_lr", 0.0001)
```

File: python/ray/tune/examples/pbt_dcgan_mnist/pbt_dcgan_mnist.py [hunk 3]
```diff
@@ -275,16 +275,11 @@ def _restore(self, checkpoint_dir):
...
+        for param_group in self.optimizerG.param_groups:
+            param_group["lr"] = new_config.get("netG_lr")
```
Comment (richardliaw) [inline python/ray/tune/examples/pbt_dcgan_mnist/pbt_dcgan_mnist.py]:
```suggestion
            param_group["lr"] = new_config.get("netG_lr", 0.0001)
```

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
jenkins test tune

**Comment by richardliaw** [comment]
jenkins test tune

**Comment by hhbyyh** [comment]
Thanks for helping review!

**Pull request merged**